lib/readyset/health/healthchecker.rb

```diff
@@ -12,14 +12,16 @@ module Health
     class Healthchecker
       UNHEALTHY_ERRORS = [::PG::UnableToSend, ::PG::ConnectionBad].freeze
 
-      def initialize(healthcheck_interval:, error_window_period:, error_window_size:, shard:)
+      def initialize(config, shard:)
         @healthy = Concurrent::AtomicBoolean.new(true)
-        @healthcheck_interval = healthcheck_interval
+        @healthcheck_interval = config.healthcheck_interval!
```

Comment:
**_question_** _(non-blocking)_ Where did the change to a `bang!` method emerge from?

Reply:
@helpotters The bang comes from `ActiveSupport::OrderedOptions`: https://api.rubyonrails.org/classes/ActiveSupport/OrderedOptions.html

Using a bang throws an exception if that option is not set